### ISSUE TITLE
x/tools/gopls: document setup required for working with the std library

### ISSUE THREAD

**Issue by katiehockman** [golang/go#38603]
Find All References in VSCode is only able to find references within the same package. For example, clicking Find All References for pkix.Name in go/src/crypto/x509/pkix/pkix.go only finds 7 references to the type, although if you use search you can see there are several more than that.

settings.json:
{
    "go.alternateTools": {
        "go": "\~/godev"
    },
    "go.toolsGopath": "\~/.vscode/godev",
    "html.format.enable": false,
    "editor.codeActionsOnSave": {
        "source.organizeImports": false
    },
    "gopls": {
        "env": {
            "PATH": "$HOME/godev:$PATH",
        },
    },
}

gopls version
golang.org/x/tools/gopls 0.4.0
    golang.org/x/tools/gopls@v0.4.0 h1:G4+YP9kaV4dJb79J5MobyApxX493Qa6VoiTceUmxqik=

LMK if you need any logs.

**Comment by stamblerre** [comment]
What was the directory that you opened in VS Code? Right now, we limit the scope of references to the workspace root.

**Comment by katiehockman** [comment]
I opened it in go/src, so the other packages are within this workspace

**Comment by stamblerre** [comment]
Ok, thanks for confirming! I'll take a look.

**Comment by hyangah** [comment]
Is `$HOME/godev` a binary or directory?

**Comment by katiehockman** [comment]
It's a directory

**Comment by hyangah** [comment]
But it is used for `"go.alternateTools"` which should be the path to the executable I think. 

Can you please provide the output of "Go: Locate Configured Go Tools" (find from the command palette) and the beginning of the `gopls` log that prints out the `go env` results.

**Comment by katiehockman** [comment]
Ah! Yep that was it. Thank you.

Is this all documented somewhere? It's still pretty difficult to work in the standard library, and this just silently failed partially, without giving any error message.

**Comment by stamblerre** [comment]
The docs for `"go.alternateTools"` should pop up when you add that configuration:

<img width="936" alt="Screen Shot 2020-04-22 at 7 33 04 PM" src="https://user-images.githubusercontent.com/5856771/80043613-32e2a680-84d0-11ea-8abe-d0feff3c9916.png">

I'm surprised that that fixed the issue (unless you are using Go Nightly) because the fix for https://github.com/microsoft/vscode-go/issues/3163 hasn't been released yet, and you have the workaround in your settings.

**Comment by katiehockman** [comment]
I was working with @FiloSottile on this config (I borrowed it from him) so I wasn't very familiar with all of these settings. He helped me set this up because I was unaware that I needed to redirect gopls to a different goroot when working in the standard library. As far as I know, this isn't in the official docs?

I don't think I'm running Go nightly, and not sure how to check, but can investigate if needed!

I'm mostly hoping for some documentation about how to set up gopls to work with the standard library, since it was very non-obvious to me (everything was working great, except for a few features that silently failed - like Find References).
